### DISCUSSION TITLE
Error with v3.4.3

### DISCUSSION THREAD

**Discussion by DrMarkX** [Snow-Shell/servicenow-powershell#222] [Q&A]
From my question #192 I am now getting several 'ID' errors when using this line.
`Get-ServiceNowRecord -Table sc_req_item -filter @('number', '-in', ($tasks.'request_item.number'[0..48] -join ',')) -IncludeCustomVariable -Property number -IncludeTotalCount`

After several iterations of that error, the correct output is displayed.

**Error**
Get-ServiceNowRecord : Cannot validate argument on parameter 'ID'. Id must be either a 32 character alphanumeric, ServiceNow sysid, or prefix/id, ServiceNow number.
At C:\Program Files\WindowsPowerShell\Modules\ServiceNow\3.4.3\Public\Get-ServiceNowRecord.ps1:357 char:126
+ ... on.item_option_new.reference' -ID $var.'sc_item_option.value' -Proper ...

**Comment by DrMarkX** [comment]
We downgraded to 3.4.2 and to stop the errors.

**Comment by gdbarron** [comment]
@DrMarkX could you provide more of the code you are running including setting $tasks?

**Reply by DrMarkX** [reply]
Here it is...

```
# Gets all tasks in queue
$tasks = @(Get-ServiceNowRecord -Table 'catalog task' -Filter $filter -Property number,state,short_description,sys_created_on,sys_updated_on,request_item.number,sys_id -IncludeTotalCount)

if(!$tasks) { Break }

if($tasks.count -eq 1) { $allRitms = @(Get-ServiceNowRecord -Table sc_req_item -filter @('number', '-eq', $tasks.'request_item.number') -IncludeCustomVariable -New -Property number) }
else {
    # Creates counting variables to split tickets up to 50 per loop
    $iter = [Math]::Ceiling($tasks.count/50)
    if($tasks.count -gt 50) { $const = $tce = [Math]::Floor($tasks.count / $iter) }
    else { $const = $tce = $tasks.count - 1 }
    $tcs = $cntr = 0

    # Loops through groups of tasks to get RITM data
    while($const -lt ($tasks.count - $cntr)) {
        Write-Host "Getting RITM records $($tcs + 1) - $($tce + 1)"
        $cntr = $cntr + ($tce - $tcs + 1)
        $ritms = @(Get-ServiceNowRecord -Table sc_req_item -filter @('number', '-in', ($tasks.'request_item.number'[$tcs..$tce] -join ',')) -IncludeCustomVariable -New -Property number -IncludeTotalCount)
        $allRitms += $ritms
        $tcs = $tce + 1
        $tce = $tcs + $const
    }

    # Last group of tasks to get RITM data (if applicable)
    if(($tasks.count - $cntr) -gt 0) {
        $tce = $tasks.count - 1
        Write-Host "Getting RITM records $($tcs + 1) - $($tce + 1)"
        $ritms = @(Get-ServiceNowRecord -Table sc_req_item -filter @('number', '-in', ($tasks.'request_item.number'[$tcs..$tce] -join ',')) -IncludeCustomVariable -New -Property number -IncludeTotalCount)
        $allRitms += $ritms
    }
}
```

**Comment by DrMarkX** [comment]
I'm still getting an error with v3.4.3.  Even if I run this simple get.
```
$tasks = Get-ServiceNowRecord -Table 'Catalog Task' -Filter $filter -Property number,short_description,sys_created_on,sys_updated_on,request_item.number,sys_id,reqstr_email -IncludeTotalCount

$ritms = @(Get-ServiceNowRecord -Table sc_req_item -filter @('number', '-in', ($tasks.'request_item.number' -join ',')) -IncludeCustomVariable -New -Property number,short_description -IncludeTotalCount)
```
When I downgrade to v3.4.2 it continues to work.
Error:
Get-ServiceNowRecord : Cannot validate argument on parameter 'ID'. Id must be either a 32 character alphanumeric, ServiceNow sysid, or prefix/id, ServiceNow number.

**Reply by ReadeB** [reply]
Had the same issue with 4.0.4   Had to download the raw code from Get-ServiceNowRecord and Update-ServiceNowRecord from the latest unreleased commits and update the ps files as Administrator.   Working now.

**Reply by gdbarron** [reply]
Apologize I merged the commit, but didn't release it to the gallery.  v4.0.5 is now out there.  Thanks for the heads up @ReadeB.
